src/wallet/cbf.rs

```diff
@@ -0,0 +1,88 @@
+use std::{collections::HashMap, net::SocketAddr, path::PathBuf, thread, time::Duration};
+use std::cell::Cell;
+
+use nakamoto::{
+    client::{
+        chan::Receiver,
+        Client,
+        Config,
+        Event,
+        Handle as ClientHandle,
+        handle::Handle,
+    },
+    net::poll::Waker,
+};
+use nakamoto::chain::Transaction;
+use nakamoto::p2p::fsm::fees::FeeEstimate;
+
+type Reactor = nakamoto::net::poll::Reactor<std::net::TcpStream>;
+
+pub struct CbfBlockchain {
+    receiver: Receiver<Event>,
+    client_handle: ClientHandle<Waker>,
+    timeout: Duration,
+    fee_data: Cell<HashMap<u32, FeeEstimate>>,
+    broadcasted_txs: Cell<Vec<Transaction>>,
+    last_sync_height: Cell<u32>,
+}
+
+pub enum CbfSyncError {
+    FilterHeaderRetrievalError,
+    BlockFilterDownloadError,
+    TransactionRetrievalError,
+}
+
+impl From<nakamoto::client::Error> for CbfSyncError {
+    fn from(_: nakamoto::client::Error) -> Self {
+        CbfSyncError::FilterHeaderRetrievalError
+    }
+}
+
+impl CbfBlockchain {
+    pub fn new(
+        network: bitcoin::Network,
+        datadir: Option<PathBuf>,
+        peers: Vec<SocketAddr>,
+    ) -> Result<Self, CbfSyncError> {
+        let root = if let Some(dir) = datadir {
+            dir
+        } else {
+            PathBuf::from(std::env::var("HOME").unwrap_or_default())
```

Comment:
Home is a bad place for this. We need a better location for the default data_dir. It should be `get_taker_dir().join("cbf")` or similar.